reana_client/openapi_connections/reana_server.json

```diff
@@ -585,20 +585,36 @@
           },
           {
             "description": "Required. New workflow status.",
-            "in": "body",
+            "in": "query",
             "name": "status",
             "required": true,
-            "schema": {
-              "description": "Required. New status.",
-              "type": "string"
-            }
+            "type": "string"
           },
           {
             "description": "Required. The API access_token of workflow owner.",
             "in": "query",
             "name": "access_token",
             "required": true,
             "type": "string"
+          },
+          {
+            "description": "Optional. Extra parameters for workflow status.",
+            "in": "body",
+            "name": "parameters",
+            "required": false,
+            "schema": {
+              "properties": {
+                "input_parameters": {
```

Comment:
Is this schema being used? I just see a free dict here:
```console
    parsed_parameters = {'parameters':
                         dict(p.split('=') for p in parameters)}
```
If this is the desired schema, we have to think exactly where we want each parameter to end up and not mix responsibilities. As I see in [`RWES` PR](https://github.com/reanahub/reana-workflow-engine-serial/pull/32), this parameters are not just "parameters", they are "workflow_engine_parameters" as in `cwltool` could be the `--timestamps` flag (what I guess `run_parameters` means) but they are passed as [workflow parameters](https://github.com/reanahub/reana-workflow-controller/blob/master/docs/openapi.json#L220-L222).

If we want also the "workflow_parameters" (i.e. in demo [root6-roofit `events`](https://github.com/reanahub/reana-demo-root6-roofit/blob/master/workflow/cwl/input.yml#L1)) (what I guess `input_parameters` are) to be overwritten by this call, then we have to update the database since [they are written to the DB at creation time](https://github.com/reanahub/reana-workflow-controller/blob/master/reana_workflow_controller/rest.py#L295).

In any case they are two different things we should differentiate. To keep it simple, it the `PUT /api/workflows/{workflow_id_or_name}/status` I would for now, only pass "workflow_engine_parameters" and add a new parameter to the [Celery task](https://github.com/reanahub/reana-workflow-engine-serial/pull/32/files#diff-6d29ceb19895382845929513c9bc9099R86) and this one would be the one representing workflow engine parameters.

Reply:
The schema needs updating, regarding the differentiation I can see input parameters (number of events for example) and workflow run parameters (caching true|false), the schema was created with these 2 types, but I went with a single dictionary for now, to keep it short.
We should ticketize this splitting parameters into 2, but since now only the `CACHING=true|false` is read from them we can use it as is.

Reply:
Makes sense 👍 . So for now we implement this just with workflow engine parameters and we add this additional parameter in the task (`engine_parameters` or similar). Later we think about how we want to handle the `workflow_parameters` (I will create issue from this discussion).

I would emphasize in those two names or similar since: one, the `workflow_parameters` is as  [we have modeled it](https://github.com/reanahub/reana-db/blob/master/reana_db/models.py#L79-L90) (or `input_parameters` as it is common terminology in `yadage` and `CWL`), and second the workflow engine parameters (related to https://github.com/reanahub/reana/pull/73#discussion_r207264526 dicussion), since they are literally parameters for the concrete workflow engine implementation (i.e. [`cwltool` ](https://manpages.debian.org/testing/cwltool/cwltool.1.en.html) ones or [`yadage`](https://yadage.readthedocs.io/en/latest/executingworkflows.html#example) ones).